pkg/menus/menus.go

```diff
@@ -0,0 +1,162 @@
+package menus
+
+import (
+	"fmt"
+	"log"
+	"os"
+	"strconv"
+
+	ui "github.com/gizak/termui/v3"
+	"github.com/gizak/termui/v3/widgets"
+)
+
+var menuWidth = 50
+
+// Entry contains all the information needed for a boot entry
+type Entry interface {
+	// Label is the string will show in menu
+	Label() string
+	// IsDefault is true means this entry will be hit by default, if there is many default choise, the first on in the list will be choose. the rest will be ignored.
+	IsDefault() bool
+}
+
+// DisplayMenu will present all entries into a menu with numbers, so that user can choose from them.
+func DisplayMenu(menuTitle string, introwords string, location int, entries []Entry) (int, error) {
```

Comment:
Instead of returning a number here, why not return the chosen entry? I'm not sure if it'd be better, just wondering

Reply:
because the entry do not contain all information I want, for example , a entry do not contain the path to the chosen iso. so I need the index  instead of entry

Reply:
I would argue you need to put more information in the entry, if it does not have enough information.

Reply:
but If I put extra information like path to iso in the entry, other kind of option may nor have this attribute like download option. and download option will need other attribute like download link etc. that will make entry struct too redundancy I guess

Reply:
> I would argue you need to put more information in the entry, if it does not have enough information.

but If I put extra information like path to iso in the entry, other kind of option may nor have this attribute like download option. and download option will need other attribute like download link etc. that will make entry struct too redundancy I guess

Reply:
@rminnich

Reply:
See my other comment for one potential way to resolve this